x-pack/plugins/search_playground/server/lib/conversational_chain.test.ts

```diff
@@ -470,102 +481,57 @@ describe('conversational chain', () => {
       const input = {
         context: 'This is a test context.',
         question: 'This is a test question.',
         chat_history: 'This is a test chat history.',
       };
+      jest.spyOn(prompt, 'format');
+      const result = await contextLimitCheck(undefined, prompt)(input);
 
-      const data = new experimental_StreamData();
-      const appendMessageAnnotationSpy = jest.spyOn(data, 'appendMessageAnnotation');
-
-      const result = await clipContext(undefined, prompt, data)(input);
-      expect(result).toEqual(input);
-      expect(appendMessageAnnotationSpy).not.toHaveBeenCalled();
+      expect(result).toBe(input);
+      expect(prompt.format).not.toHaveBeenCalled();
     });
 
-    it('should not clip context if within modelLimit', async () => {
+    it('should return the input if within modelLimit', async () => {
       const input = {
         context: 'This is a test context.',
         question: 'This is a test question.',
         chat_history: 'This is a test chat history.',
       };
-      const data = new experimental_StreamData();
-      const appendMessageAnnotationSpy = jest.spyOn(data, 'appendMessageAnnotation');
-      const result = await clipContext(10000, prompt, data)(input);
+      jest.spyOn(prompt, 'format');
+      const result = await contextLimitCheck(10000, prompt)(input);
       expect(result).toEqual(input);
-      expect(appendMessageAnnotationSpy).not.toHaveBeenCalled();
+      expect(prompt.format).toHaveBeenCalledWith(input);
     });
 
     it('should clip context if exceeds modelLimit', async () => {
+      expect.assertions(1);
       const input = {
         context: 'This is a test context.\nThis is another line.\nAnd another one.',
         question: 'This is a test question.',
         chat_history: 'This is a test chat history.',
       };
-      const data = new experimental_StreamData();
-      const appendMessageAnnotationSpy = jest.spyOn(data, 'appendMessageAnnotation');
-      const result = await clipContext(33, prompt, data)(input);
-      expect(result.context).toBe('This is a test context.\nThis is another line.');
-      expect(appendMessageAnnotationSpy).toHaveBeenCalledWith({
-        type: 'context_clipped',
-        count: 4,
-      });
-    });
-
-    it('exit when context becomes empty', async () => {
-      const input = {
-        context: 'This is a test context.\nThis is another line.\nAnd another one.',
-        question: 'This is a test question.',
-        chat_history: 'This is a test chat history.',
-      };
-      const data = new experimental_StreamData();
-      const appendMessageAnnotationSpy = jest.spyOn(data, 'appendMessageAnnotation');
-      const result = await clipContext(1, prompt, data)(input);
-      expect(result.context).toBe('');
-      expect(appendMessageAnnotationSpy).toHaveBeenCalledWith({
-        type: 'context_clipped',
-        count: 15,
-      });
+      try {
+        await contextLimitCheck(33, prompt)(input);
```

Comment:
Nitpick: rather than test the method toString, check that it throws an exception like 

```
expect(failingAsyncTest())
    .rejects
    .toThrow('I should fail');
```

Reply:
Good catch, thank you
https://jestjs.io/docs/asynchronous#asyncawait